src/openassetio-core/tests/utils/PrintableTest.cpp

```diff
@@ -0,0 +1,191 @@
+                                 "aStringTraitProperty", openassetio::Str("a string"));
+    traitsData->setTraitProperty("a.long.namespaced.trait.that.goes.on.and.on.and.on",
+                                 "aBoolTraitProperty", true);
+    traitsData->addTrait("a.trait.with.no.properties");
+
+    WHEN("An InfoDictionary is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintableByCharacters(infoDictionary, "{`key2`: False, `key1`: 'value1'}");
+      }
+    }
+
+    WHEN("A TraitSet is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintableByCharacters(traitSet, "{'trait2', 'trait1'}");
+      }
+    }
+
+    WHEN("TraitSets are printed") {
+      THEN("They can be printed via fmt and <<") {
+        checkBasicPrintableByCharacters(traitSets, "[{'trait2', 'trait1'}, {'trait4', 'trait3'}]");
+      }
+    }
+
+    WHEN("A BatchElementError is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintable(batchElementError, "invalidTraitSet: Invalid trait set");
+      }
+    }
+
+    WHEN("An EntityReference is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintable(entityReference, "test:///an_entity_reference");
+      }
+    }
+
+    WHEN("EntityReferences are printed") {
+      THEN("They can be printed via fmt and <<") {
+        checkBasicPrintable(entityReferences,
+                            "['test:///an_entity_reference_1', 'test:///an_entity_reference_2']");
+      }
+    }
+
+    WHEN("An Identifier is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintable(identifier, "an identifier");
+      }
+    }
+
+    WHEN("A Str is printed") {
+      THEN("It can be printed via fmt and <<") { checkBasicPrintable(str, "example string"); }
+    }
+
+    WHEN("A StrMap is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintableByCharacters(strMap, "{'key2': 'value2', 'key1': 'value1'}");
+      }
+    }
+
+    WHEN("A Capability is printed") {
+      THEN("It can be printed via fmt and <<") { checkBasicPrintable(capability, "publishing"); }
+    }
+
+    WHEN("A Context is printed") {
+      THEN("It can be printed via fmt and <<") {
+        // No closing brace on purpose, to account for variant managerState memory address
+        checkBasicPrintableContains(*context,
+                                    R"({'locale': {'aTrait': {'aIntTraitProperty': 2}}, )"
+                                    R"('managerState': 0x)");
+        checkBasicPrintableContains(context, R"('locale': {'aTrait': {'aIntTraitProperty': 2}}, )"
+                                             R"('managerState': 0x)");
+      }
+    }
+
+    WHEN("A TraitsData is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintableByCharacters(
+            *traitsData,
```

Comment:
[fix] Remove `*` prefix, otherwise we're just asserting the same thing twice.

Reply:
You mean because the pointer overloads call through the the base implementation so there's no need to test both?
My thought was it was worth testing that a pointer and value overload both could be called publicly, as we're providing them as essentially public interface.

Reply:
Yeah, but you're dereferencing the pointer before passing it to `checkBasicPrintableByCharacters`, so its not testing stringification of the pointer.

Reply:
Right yeah, the actual function is called inside the helper duh.

Reply:
... you're gonna have to explain this one to me face to face. The idea is that you have a pointer, and you test both the stringificaton of the actual data and the pointer, by deferencing the pointer to test the value-date, and then immediately below, not referencing it to test it as a pointer.

Reply:
Nope you're right, I completely misread :facepalm: